code/modules/projectiles/guns/energy.dm

```diff
@@ -146,6 +146,12 @@
 		flashlight_overlay.pixel_x = flight_x_offset
 		flashlight_overlay.pixel_y = flight_y_offset
 		add_overlay(flashlight_overlay)
+	if(bayonet && can_knife)
+		var/iconK = "bayonet"

```

Comment:
@RemieRichards uhhhhhhhh we don't have sprites for all that, and i only have a basic bayonet sprites, so I guess I'll do that and have it detect if there's a sprite for it and if not use the generic one?